vcr/statuslist/store_test.go

```diff
@@ -0,0 +1,256 @@
+				Where("status_list_credential = ?", statusListCredential).
+				Update("last_issued_index", statuslist2021.MaxBitstringIndex)
+			statusListCredential, _ = toStatusListCredential(aliceDID, 2) // now expect page 2 to be used
+
+			entry, err = s.Create(nil, aliceDID, StatusPurposeRevocation)
+
+			assert.NoError(t, err)
+			require.NotNil(t, entry)
+
+			assert.Equal(t, statusListCredential, entry.StatusListCredential)
+			assert.Equal(t, "0", entry.StatusListIndex)
+			assert.Equal(t, fmt.Sprintf("%s#0", statusListCredential), entry.ID)
+		})
+		t.Run("second issuer", func(t *testing.T) {
+			entry, err = s.Create(nil, bobDID, StatusPurposeRevocation)
+
+			assert.NoError(t, err)
+			require.NotNil(t, entry)
+			assert.Equal(t, "0", entry.StatusListIndex)
+			// alice#1, alice#2, bob#1; fails if only part of the 'ok' block is executed
+			assert.Equal(t, int64(3), s.db.Find(&[]statusListCredentialRecord{}).RowsAffected)
+		})
+	})
+	t.Run("error - unsupported purpose", func(t *testing.T) {
+		entry, err = s.Create(nil, aliceDID, statusPurposeSuspension)
+		assert.ErrorIs(t, err, errUnsupportedPurpose)
+		assert.Nil(t, entry)
+	})
+	t.Run("error - unsupported ID method", func(t *testing.T) {
+		entry, err = s.Create(nil, did.MustParseDID("did:nuts:123"), StatusPurposeRevocation)
+		assert.EqualError(t, err, "status list: unsupported DID method: nuts")
+		assert.Nil(t, entry)
+	})
+	t.Run("optimistic locking", func(t *testing.T) {
+		t.Run("gorm translates errors", func(t *testing.T) {
+			// without this the lock conditions are not handled correctly
+			assert.True(t, s.db.Config.TranslateError)
+		})
+		t.Run("trigger gorm.ErrRecordNotFound on UPDATE of LastIssuedIndex", func(t *testing.T) {
+			t.SkipNow()
+			// This test is skipped since in needs manual evaluation.
+			// Run this test if the logic around optimistic locking has been changed.
+
+			// After running this test check the console to verify that it has log messages of the form:
+			// page already creation
+			// 		.../nuts-node/vcr/issuer/statuslist2021_store.go:179 duplicated key not allowed
+			//		[1.389ms] [rows:0] INSERT INTO `status_list_credentials` (`status_list_credential`,`issuer`,`page`,`last_issued_index`) VALUES ("https://example.com/iam/alice/statuslist/1","did:web:example.com:iam:alice",1,0)
+			//
+			// last index already updated
+			// 		.../nuts-node/vcr/issuer/statuslist2021_store.go:203 SLOW SQL >= 200ms
+			//		[260.812ms] [rows:0] UPDATE `status_list_credentials` SET `last_issued_index`=491 WHERE status_list_credential = "https://example.com/iam/alice/statuslist/1" AND last_issued_index = 490
+			//
+			// If this does not produce the required messages, make sure logging is on, or change values 'numWorkers' and 'numJobs'
+
+			worker := func(jobs <-chan int, results chan<- bool) {
+				for range jobs {
+					_, _ = s.Create(nil, aliceDID, StatusPurposeRevocation)
+					results <- true
+				}
+			}
+
+			const numWorkers = 10
+			const numJobs = 4000
```

Comment:
this is a complicated way to implement a wait group?

Reply:
Wait groups are used to ensure all routines are closed. 
Changing this to a wait group would fire a new go routine for every job, which is presumably slower than letting active routines pick new jobs as long as there are any?